### DISCUSSION TITLE
setQueryData() => {copy: false}

### DISCUSSION THREAD

**Discussion by utftu** [TanStack/query#1860] [Ideas]
Now the `setData()` method looks like this, 
```js
// Use prev data if an isDataEqual function is defined and returns `true`
if (this.options.isDataEqual?.(prevData, data)) {
  data = prevData as TData
} else if (this.options.structuralSharing !== false) {
  // Structurally share data between prev and new data if needed
  data = replaceEqualDeep(prevData, data)
}
```
but I would like to be able to use the `setQueryData()` method to set data without creating a deep copy by the library and disabling the structural sharing.

I suggest adding the `copy` field to the existing config 
```ts
interface SetDataOptions {
  updatedAt?: number,
  copy?: boolean // new field
}
```
, which will mean that there is no need to create a deep copy. Then the `setData()` code should look like this
```js
// Use prev data if an isDataEqual function is defined and returns `true`
if (this.options.isDataEqual?.(prevData, data)) {
  data = prevData as TData
} else if (this.options.structuralSharing !== false && options.copy !== false) {
  // Structurally share data between prev and new data if needed
  data = replaceEqualDeep(prevData, data)
}
```

and the call to `setQueryData()` will look like this 
```js
setQueryData(queryKey, data, {copy: false})
```

### What is it for?

Now I am developing a library for normalizing the cache and a wrapper for react-query and I would like to eliminate the need to create a deep copy of the data, not counting that it wastes crumbs of performance, this behavior also does not allow having the same entities in different queries

**Comment by TkDodo** [comment]
Not sure that I fully understand the use-case, but why don't you just set `structuralSharing: false` ?
